File: 0217-contains-duplicate/0217-contains-duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        int n = nums.size();

        unordered_set<int> map;

        for (int num : nums) {
            if (map.find(num) != map.end()) {
                return true;
            }
            map.insert(num);
        }
        return false;
    }
};

File: 0217-contains-duplicate/0217-contains-duplicate.py
class Solution(object):
    def containsDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        n = len(nums)

        nums.sort() 

        for i in range(1,n) :
            if(nums[i] == nums[i-1]) :
                return True
        return False